python/llm/test/inference_gpu/test_optimize_model.py

```diff
@@ -56,5 +80,115 @@ def test_optimize_model(Model, Tokenizer, model_path):
     assert any(diff) is False
 
 
+@pytest.mark.parametrize('Model, Tokenizer, model_path',[
+    (AutoModelForCausalLM, AutoTokenizer, os.environ.get('LLAMA_ORIGIN_PATH'))
+])
+
+def test_optimize_llama_model(Model, Tokenizer, model_path):
+    tokenizer = Tokenizer.from_pretrained(model_path, trust_remote_code=True)
+    input_ids = tokenizer.encode(prompt, return_tensors="pt").to(device)
+
+    model = Model.from_pretrained(model_path,
+                                load_in_4bit=True,
+                                optimize_model=False,
+                                trust_remote_code=True)
+    model = model.to(device)
+
+    for layer_name, layer_module in model.named_modules():
+        if layer_name == "model.layers.31.self_attn":
+            layer_module.register_forward_pre_hook(
+                lambda module, input: pre_hook(module, input))
+            layer_module.register_forward_hook(
+                lambda module, output, layer_name=layer_name: forward_hook(module, 
+                                                                           output, layer_name))
+    logits_base_model = (model(input_ids)).logits
+    layer_tensor = layer_outputs.pop()
```

Comment:
why need to pop here?

Reply:
You need to add some doc here. It is a singleton right, you are taking the last and only item right?

Reply:
yes, it is, will add a doc.